### PR TITLE
Jakarta Concurrency PR

### PR DESCRIPTION
**Pull request by smillidge**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [ ] Generated IP Log
   - [ ] Email to PMC
   - [ ] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/2.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-concurrency-tck-2.0.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/concurrency-api/issues/130
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by kwsutter** [comment]
@karianna and Marcelo Ancelmo (github id?) are the assigned Mentors for this Specification PR.  Thanks!

**Comment by ivargrimstad** [comment]
Release date should be on front page.

The following fields on the spec license page (both pdf and HTML) (page 1) should be updated to:
Status: **Final Release**
Release: **<the release date>**

File: concurrency/2.0/_index.md [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
   * [Jakarta EE Platform 9 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan)
-* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0-RC2.pdf) (PDF)
-* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0-RC2.html) (HTML)
+* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0.pdf) (PDF)
+* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0.html) (HTML)
 * [Jakarta Concurrency 2.0 Javadoc](./apidocs)
 * [Jakarta Concurrency 2.0 TCK]()([sig](),[sha](),[pub]())
 * Maven coordinates
-  * [jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/2.0.0-RC2/jakarta.enterprise.concurrent-api-2.0.0-RC2.jar)
+  * [jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/2.0.0/jakarta.enterprise.concurrent-api-2.0.0.jar)
```
Comment (ivargrimstad) [inline concurrency/2.0/_index.md:15]: Remove -RC2 from link text

File: concurrency/2.0/_index.md [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
 Jakarta Concurrency provides a specification for using concurrency from application components without compromising container integrity while still preserving the Jakarta EE platform's fundamental benefits.
 
 * [Jakarta Concurrency 2.0 Release Record](https://projects.eclipse.org/projects/ee4j.cu/releases/2.0.0)
   * [Jakarta EE Platform 9 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan)
-* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0-RC2.pdf) (PDF)
-* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0-RC2.html) (HTML)
+* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0.pdf) (PDF)
+* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0.html) (HTML)
 * [Jakarta Concurrency 2.0 Javadoc](./apidocs)
 * [Jakarta Concurrency 2.0 TCK]()([sig](),[sha](),[pub]())
```
Comment (ivargrimstad) [inline concurrency/2.0/_index.md:13]: TCK links need to be added:
- https://download.eclipse.org/jakartaee/concurrency/2.0/concurrency-tck-2.0.0.zip 
- sig: https://download.eclipse.org/jakartaee/concurrency/2.0/concurrency-tck-2.0.0.zip.sig 
- sha: https://download.eclipse.org/jakartaee/concurrency/2.0/concurrency-tck-2.0.0.zip.sha256 
- pub: https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub

File: concurrency/2.0/_index.md [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
 * Maven coordinates
-  * [jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/2.0.0-RC2/jakarta.enterprise.concurrent-api-2.0.0-RC2.jar)
+  * [jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/2.0.0/jakarta.enterprise.concurrent-api-2.0.0.jar)
 
 
 # Compatible Implementations
 
-* [Jakarta Concurrency CI 2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/org/glassfish/jakarta.enterprise.concurrent/2.0.0-RC2/jakarta.enterprise.concurrent-2.0.0-RC2.jar)
+* [Jakarta Concurrency CI 2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/org/glassfish/jakarta.enterprise.concurrent/2.0.0/jakarta.enterprise.concurrent-2.0.0.jar)
```
Comment (ivargrimstad) [inline concurrency/2.0/_index.md:20]: Remove -RC2 from link name

**Review by ivargrimstad — CHANGES_REQUESTED** [review]

**Comment by smillidge** [comment]
> Release date should be on front page.
> 
> The following fields on the spec license page (both pdf and HTML) (page 1) should be updated to:
> Status: **Final Release**
> Release:

The pom says it should be blank for final? Has that changed now?

**Comment by ivargrimstad** [comment]
> > Release date should be on front page.
> > The following fields on the spec license page (both pdf and HTML) (page 1) should be updated to:
> > Status: **Final Release**
> > Release:
> 
> The pom says it should be blank for final? Has that changed now?

When we staged MVC last week, the field should be empty in the POM. We discovered that our release script must be run from the master branch for it to work correctly with these parameters. The jenkins job didn't use the release branch parameter at all.

**Review by karianna — CHANGES_REQUESTED** [review]
It all looks good to me so far (apart from the obvious items that are still to go on the checklist, which I'm ticking off as we speak)

I did notice that `Index page {spec}/_index.md following template` had been checked off, but I didn't see that file in the PR...

**Comment by karianna** [comment]
@smillidge - This is getting close - I think a next major step is to provide The URL of the compatibility certification request issue (I assume in the concurrency-ri project at ee4j

**Comment by ivargrimstad** [comment]
> I did notice that `Index page {spec}/_index.md following template` had been checked off, but I didn't see that file in the PR...

That file does not need any changes for this release as it was added for the previous release. The checklist item is a rest from that...

**Comment by marceloancelmo** [comment]
> the assigned Mento

My github ID is @marceloancelmo, can you please add me so I can help on the reviews

**Comment by smillidge** [comment]
@marceloancelmo I tried to add you as a reviewer but I can't

**Comment by smillidge** [comment]
I've updated the Spec document to say Final Release

**Comment by ivargrimstad** [comment]
> @marceloancelmo I tried to add you as a reviewer but I can't

Only Specification Committee members can be assigned as reviewers

**Comment by ivargrimstad** [comment]
> I've updated the Spec document to say Final Release

Great!
Release Date is still missing...

File: concurrency/2.0/_index.md [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
 
 * [Jakarta Concurrency 2.0 Release Record](https://projects.eclipse.org/projects/ee4j.cu/releases/2.0.0)
   * [Jakarta EE Platform 9 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan)
-* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0-RC2.pdf) (PDF)
-* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0-RC2.html) (HTML)
+* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0.pdf) (PDF)
+* [Jakarta Concurrency 2.0 Specification Document](./concurrency-spec-2.0.html) (HTML)
 * [Jakarta Concurrency 2.0 Javadoc](./apidocs)
-* [Jakarta Concurrency 2.0 TCK]()([sig](),[sha](),[pub]())
+* [Jakarta Concurrency 2.0 TCK](https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/staged-900/jakarta-concurrency-tck-2.0.0.zip)([sig](),[sha](),[pub]())
```
Comment (ivargrimstad) [inline concurrency/2.0/_index.md:13]: These links must be:
tck: https://download.eclipse.org/jakartaee/concuerrency/2.0/jakarta-concurrency-tck-2.0.0.zip
sig: https://download.eclipse.org/jakartaee/concuerrency/2.0/jakarta-concurrency-tck-2.0.0.zip.sig
sha: https://download.eclipse.org/jakartaee/concuerrency/2.0/jakarta-concurrency-tck-2.0.0.zip.sha256
pub: https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub

Comment (smillidge) [reply concurrency/2.0/_index.md:13]: OK will do and try to work out how to get the date into the document.

Comment (ivargrimstad) [reply concurrency/2.0/_index.md:13]: In MVC, we set the following configuration attribute in the execution definition of the asciidoctor-plugin:
                 <revdate>${revisiondate}</revdate>

Complete example here: https://github.com/eclipse-ee4j/mvc-api/blob/master/spec/pom.xml

Comment (smillidge) [reply concurrency/2.0/_index.md:13]: Thanks that helped and the spec should have the date in now

Comment (smillidge) [reply concurrency/2.0/_index.md]: TCK links also updated

**Review by ivargrimstad — APPROVED** [review]

**Comment by ivargrimstad** [comment]
Only thing left now is the SHA-256 in the Certification request+TCK results and merge the TCK results PR to the GlassFish web page. Then we're good to go.

**Comment by smillidge** [comment]
SHA now corrected

**Comment by ivargrimstad** [comment]
I have started the ballot. Please proceed with the steps involving the EMO and the PMC in order to release.

File: concurrency/2.0/_index.md [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
 * [Jakarta Concurrency 2.0 Javadoc](./apidocs)
-* [Jakarta Concurrency 2.0 TCK]()([sig](),[sha](),[pub]())
+* [Jakarta Concurrency 2.0 TCK](https://download.eclipse.org/jakartaee/concurrency/2.0/jakarta-concurrency-tck-2.0.0.zip)([sig](https://download.eclipse.org/jakartaee/concurrency/2.0/jakarta-concurrency-tck-2.0.0.zip.sig),[sha](https://download.eclipse.org/jakartaee/concurrency/2.0/jakarta-concurrency-tck-2.0.0.zip.sha256),[pub](https://raw.githubusercontent.com/jakartaee/specification-committee/master/jakartaee-spec-committee.pub))
 * Maven coordinates
-  * [jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/2.0.0-RC2/jakarta.enterprise.concurrent-api-2.0.0-RC2.jar)
+  * [jakarta.enterprise.concurrent:jakarta.enterprise.concurrent-api:2.0.0](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/jakarta/enterprise/concurrent/jakarta.enterprise.concurrent-api/2.0.0/jakarta.enterprise.concurrent-api-2.0.0.jar)
```
Comment (kazumura) [inline concurrency/2.0/_index.md:16]: This looks like staging repo. It should be "https://search.maven.org/artifact/...".

File: concurrency/2.0/_index.md [hunk 6]
```diff
@@ -1,23 +1,23 @@
...
 
 
 # Compatible Implementations
 
-* [Jakarta Concurrency CI 2.0.0-RC2](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/org/glassfish/jakarta.enterprise.concurrent/2.0.0-RC2/jakarta.enterprise.concurrent-2.0.0-RC2.jar)
+* [Jakarta Concurrency CI 2.0.0](https://jakarta.oss.sonatype.org/service/local/repo_groups/staging/content/org/glassfish/jakarta.enterprise.concurrent/2.0.0/jakarta.enterprise.concurrent-2.0.0.jar)
```
Comment (kazumura) [inline concurrency/2.0/_index.md:21]: CI is different from the compatibility certification request ("https://github.com/eclipse-ee4j/concurrency-api/issues/130")

File: concurrency/2.0/_index.md (hunk 5 above)
Comment (ivargrimstad) [reply concurrency/2.0/_index.md:16]: Good catch @kazumura!
@smillidge Should be updated before we merge.

File: concurrency/2.0/_index.md (hunk 6 above)
Comment (ivargrimstad) [reply concurrency/2.0/_index.md:21]: It is concurrent version 2.0.0 that will be included in the final release of GlassFish 6 according to the pom file (https://github.com/eclipse-ee4j/glassfish/blob/master/appserver/pom.xml). I would say that the CI version linked from there should be a link to the released version and not necessarily be the milestone used in the certification request. Should not be a staged version though

Comment (smillidge) [reply concurrency/2.0/_index.md:21]: Not sure to do with the comment? Do we want the version that passed the TCK or do we want the final version incorporated into a to be released GlassFish. It always confuses me these links etc. to things that don't exist yet!

Comment (ivargrimstad) [reply concurrency/2.0/_index.md:21]: Agree, it is confusing. Here's my take:

- The link in the _index.md file should be to the download location of the final release of the compatible implementation. Could be a link to something that is not there yet
- The link in the certification request should be to the download location of the specific version (milestone, release candidate, etc.) that was used to pass the TCK. It should not be a snapshot, or point to a location or file that will be removed later. It should be persistent.

Comment (smillidge) [reply concurrency/2.0/_index.md:21]: So we should be releasing RCs to maven central?

Comment (ivargrimstad) [reply concurrency/2.0/_index.md:21]: If you're certifying on an RC, I think you should. Unless you can provide a separate download location for it that does not go away.

Comment (kwsutter) [reply concurrency/2.0/_index.md:21]: Did you use Glassfish 6.0-M1 to do the compatibility testing for Concurrency 2.0?  If so, that is a permanent link.

Comment (smillidge) [reply concurrency/2.0/_index.md:21]: yes I did use GlassFish M1

**Comment by starksm64** [comment]
In looking at the specifications preview page on jakarta.ee, the Jakarta Concurrency CI 2.0.0 link under the Compatible Implementations section points to the staged api jar rather than some actual implementation.

**Comment by smillidge** [comment]
> In looking at the specifications preview page on jakarta.ee, the Jakarta Concurrency CI 2.0.0 link under the Compatible Implementations section points to the staged api jar rather than some actual implementation.

@starksm64 the link is to the Compatible Implementation not the api jar.

**Comment by ivargrimstad** [comment]
* On ballot completion, the specification committee mentor:
 - [ ] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
https://github.com/eclipse-ee4j/concurrency-api/issues/133

**Pull request merged**